core/src/main/scala/org/apache/spark/deploy/history/EventLogFileReaders.scala

```diff
@@ -119,7 +119,8 @@ object EventLogFileReader extends Logging {
     if (isSingleEventLog(status)) {
       Some(new SingleFileEventLogFileReader(fs, status.getPath, Option(status)))
     } else if (isRollingEventLogs(status)) {
-      if (fs.listStatus(status.getPath).exists(RollingEventLogFilesWriter.isEventLogFile)) {
+      if (fs.listStatus(status.getPath).exists(RollingEventLogFilesWriter.isEventLogFile) &&
+          fs.listStatus(status.getPath).exists(RollingEventLogFilesWriter.isAppStatusFile)) {
```

Comment:
This will result in two `listStatus`, merge them ?

Reply:
+1

Reply:
Actually yeah ...

Reply:
This is intentional because we cannot merge them, @mridulm , @LuciferYang , @HyukjinKwon .

I guess the suggestion is `.exists(RollingEventLogFilesWriter.isEventLogFile || RollingEventLogFilesWriter.isAppStatusFile)`, right? In that case, we still cannot catch a missing app status file.

Reply:
As a second thought, maybe, was the suggestion to reuse the result of `fs.listStatus(status.getPath)` as a variable? It makes senses. I'll make a PR soon.

Reply:
Here is a follow-up.
- https://github.com/apache/spark/pull/43944